filters/include/pcl/filters/impl/morphological_filter.hpp

```diff
@@ -52,157 +52,158 @@
-      {
-        Eigen::Vector3f bbox_min, bbox_max;
-        std::vector<int> pt_indices;
-        float minx = cloud_temp.points[p_idx].x - half_res;
-        float miny = cloud_temp.points[p_idx].y - half_res;
-        float minz = -std::numeric_limits<float>::max ();
-        float maxx = cloud_temp.points[p_idx].x + half_res;
-        float maxy = cloud_temp.points[p_idx].y + half_res;
-        float maxz = std::numeric_limits<float>::max ();
-        bbox_min = Eigen::Vector3f (minx, miny, minz);
-        bbox_max = Eigen::Vector3f (maxx, maxy, maxz);
-        tree.boxSearch (bbox_min, bbox_max, pt_indices);
-
-        if (!pt_indices.empty ())
+        for (std::size_t p_idx = 0; p_idx < cloud_temp.points.size (); ++p_idx)
         {
-          Eigen::Vector4f min_pt, max_pt;
-          pcl::getMinMax3D<PointT> (cloud_temp, pt_indices, min_pt, max_pt);
-
-          switch (morphological_operator)
+          Eigen::Vector3f bbox_min, bbox_max;
+          std::vector<int> pt_indices;
+          float minx = cloud_temp.points[p_idx].x - half_res;
+          float miny = cloud_temp.points[p_idx].y - half_res;
+          float minz = -std::numeric_limits<float>::max ();
+          float maxx = cloud_temp.points[p_idx].x + half_res;
+          float maxy = cloud_temp.points[p_idx].y + half_res;
+          float maxz = std::numeric_limits<float>::max ();
+          bbox_min = Eigen::Vector3f (minx, miny, minz);
+          bbox_max = Eigen::Vector3f (maxx, maxy, maxz);
+          tree.boxSearch (bbox_min, bbox_max, pt_indices);
+
+          if (!pt_indices.empty ())
           {
-            case MORPH_OPEN:
-            default:
-            {
-              cloud_out.points[p_idx].z = max_pt.z ();
-              break;
-            }
-            case MORPH_CLOSE:
+            Eigen::Vector4f min_pt, max_pt;
+            pcl::getMinMax3D<PointT> (cloud_temp, pt_indices, min_pt, max_pt);
+
+            switch (morphological_operator)
             {
-              cloud_out.points[p_idx].z = min_pt.z ();
-              break;
+              case MORPH_OPEN:
+              default:
+              {
+                cloud_out.points[p_idx].z = max_pt.z ();
+                break;
+              }
+              case MORPH_CLOSE:
+              {
+                cloud_out.points[p_idx].z = min_pt.z ();
+                break;
+              }
             }
           }
         }
+        break;
+      }
+      default:
+      {
+        PCL_ERROR ("Morphological operator is not supported!\n");
+        break;
       }
-      break;
-    }
-    default:
-    {
-      PCL_ERROR ("Morphological operator is not supported!\n");
-      break;
     }
-  }
 
-  return;
+    return;
+  }
 }
 
 #define PCL_INSTANTIATE_applyMorphologicalOperator(T) template PCL_EXPORTS void pcl::applyMorphologicalOperator<T> (const pcl::PointCloud<T>::ConstPtr &, float, const int, pcl::PointCloud<T> &);
 
 #endif  //#ifndef PCL_FILTERS_IMPL_MORPHOLOGICAL_FILTER_H_
-
```

Comment:
Please don't remove the empty line at the end of text files

Reply:
I always leave one extra line, as is best practice. This is the case here as well, I don't know why github is showing no lines, but there is exactly one empty line after the include guard. I'll delete and reinsert to see if it changes the situation.

Reply:
Sometimes mixing windows and linux style line endings is the cause. Maybe open this in vim and show the hidden chars to see if that's the case

Reply:
I think I fixed this instance, but I'll check on other PRs with vim. Thanks